File: pegasusX/apps/retailer-app-android/app/src/main/java/com/pegasusx/retailer/data/model/ClaimModels.kt
package com.pegasusx.retailer.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class RetailerClaimLine(
    val sku: String,
    val quantity: Long,
    val reason: String? = null,
    @SerialName("unit_price_minor") val unitPriceMinor: Long? = null,
    @SerialName("amount_minor") val amountMinor: Long? = null,
)

@Serializable
data class RetailerClaim(
    @SerialName("claim_id") val claimId: String,
    @SerialName("order_id") val orderId: String = "",
    @SerialName("claim_type") val claimType: String = "",
    val status: String = "",
    val description: String? = null,
    @SerialName("amount_minor") val amountMinor: Long? = null,
    val currency: String? = null,
    @SerialName("line_items") val lineItems: List<RetailerClaimLine>? = null,
    @SerialName("created_at") val createdAt: String? = null,
)

@Serializable
data class RetailerClaimsListResponse(
    val claims: List<RetailerClaim> = emptyList(),
)

@Serializable
data class ClaimEligibility(
    val eligible: Boolean = false,
    @SerialName("ends_at") val endsAt: String? = null,
    @SerialName("window_hours") val windowHours: Int = 0,
    @SerialName("hours_remaining") val hoursRemaining: Double = 0.0,
    @SerialName("policy_source") val policySource: String = "",
    @SerialName("photo_required_types") val photoRequiredTypes: List<String> = emptyList(),
    @SerialName("order_status") val orderStatus: String = "",
    val reason: String = "",
)

@Serializable
data class FileClaimEvidenceBody(
    @SerialName("evidence_type") val evidenceType: String,
    val uri: String,
    @SerialName("mime_type") val mimeType: String = "image/jpeg",
)

@Serializable
data class FileClaimLineBody(
    val sku: String,
    val quantity: Long,
    val reason: String,
)

@Serializable
data class FileClaimRequestBody(
    @SerialName("claim_type") val claimType: String,
    val description: String = "",
    @SerialName("line_items") val lineItems: List<FileClaimLineBody> = emptyList(),
    val evidences: List<FileClaimEvidenceBody> = emptyList(),
)

@Serializable
data class MediaUploadTicket(
    @SerialName("upload_url") val uploadUrl: String,
    @SerialName("public_url") val publicUrl: String? = null,
    @SerialName("image_url") val imageUrl: String? = null,
    @SerialName("content_type") val contentType: String? = null,
) {
    val resolvedPublicUrl: String
        get() = publicUrl?.takeIf { it.isNotBlank() }
            ?: imageUrl?.takeIf { it.isNotBlank() }
            ?: error("upload ticket missing public url")
}

File: pegasusX/apps/retailer-app-ios/retailerapp/retailerapp/Models/ClaimModels.swift
import Foundation

// MARK: - Logistics claims (post-delivery)

struct RetailerClaimLine: Codable, Identifiable, Hashable {
    var id: String { sku }
    let sku: String
    let quantity: Int64
    let reason: String?
    let unitPriceMinor: Int64?
    let amountMinor: Int64?

    enum CodingKeys: String, CodingKey {
        case sku
        case quantity
        case reason
        case unitPriceMinor = "unit_price_minor"
        case amountMinor = "amount_minor"
    }
}

struct RetailerClaimEvidence: Codable, Hashable {
    let evidenceType: String
    let uri: String
    let mimeType: String?

    enum CodingKeys: String, CodingKey {
        case evidenceType = "evidence_type"
        case uri
        case mimeType = "mime_type"
    }
}

struct RetailerClaim: Codable, Identifiable, Hashable {
    var id: String { claimId }
    let claimId: String
    let orderId: String
    let claimType: String
    let status: String
    let description: String?
    let amountMinor: Int64?
    let currency: String?
    let lineItems: [RetailerClaimLine]?
    let createdAt: String?

    enum CodingKeys: String, CodingKey {
        case claimId = "claim_id"
        case orderId = "order_id"
        case claimType = "claim_type"
        case status
        case description
        case amountMinor = "amount_minor"
        case currency
        case lineItems = "line_items"
        case createdAt = "created_at"
    }
}

struct RetailerClaimsListResponse: Codable {
    let claims: [RetailerClaim]
}

struct ClaimEligibility: Codable, Hashable {
    let eligible: Bool
    let endsAt: String?
    let windowHours: Int
    let hoursRemaining: Double
    let policySource: String
    let photoRequiredTypes: [String]?
    let orderStatus: String?
    let reason: String?

    enum CodingKeys: String, CodingKey {
        case eligible
        case endsAt = "ends_at"
        case windowHours = "window_hours"
        case hoursRemaining = "hours_remaining"
        case policySource = "policy_source"
        case photoRequiredTypes = "photo_required_types"
        case orderStatus = "order_status"
        case reason
    }
}

struct FileClaimEvidenceBody: Encodable {
    let evidenceType: String
    let uri: String
    let mimeType: String

    enum CodingKeys: String, CodingKey {
        case evidenceType = "evidence_type"
        case uri
        case mimeType = "mime_type"
    }
}

struct FileClaimLineBody: Encodable {
    let sku: String
    let quantity: Int64
    let reason: String
}

struct FileClaimRequestBody: Encodable {
    let claimType: String
    let description: String
    let lineItems: [FileClaimLineBody]
    let evidences: [FileClaimEvidenceBody]

    enum CodingKeys: String, CodingKey {
        case claimType = "claim_type"
        case description
        case lineItems = "line_items"
        case evidences
    }
}
